src/dawn_native/vulkan/CommandBufferVk.cpp

```diff
@@ -53,16 +53,23 @@ namespace dawn_native { namespace vulkan {
 
             region.imageSubresource.aspectMask = texture->GetVkAspectMask();
             region.imageSubresource.mipLevel = textureLocation.level;
-            region.imageSubresource.baseArrayLayer = 0;
-            region.imageSubresource.layerCount = 1;
 
-            region.imageOffset.x = textureLocation.x;
-            region.imageOffset.y = textureLocation.y;
-            region.imageOffset.z = textureLocation.z;
-
-            region.imageExtent.width = textureLocation.width;
-            region.imageExtent.height = textureLocation.height;
-            region.imageExtent.depth = textureLocation.depth;
+            switch (texture->GetDimension()) {
+                case dawn::TextureDimension::e2D:
+                    region.imageSubresource.baseArrayLayer = textureLocation.z;
```

Comment:
Like in [1] this should probably be a separate argument, see the proposal for texture copies [here](https://github.com/gpuweb/gpuweb/issues/69).

Reply:
I once thought as there were no "3D array" textures, maybe we could avoid using one more variable here. Sorry that I failed to catch this key point in this proposal.

Reply:
No worries, I don't think there are 3D array textures in any API (and actually you might be able to convert 2D array to a 3D texture view in some conditions) so things are a bit confusing.